applications: nrf_desktop: Add DFU chapter to readme (#2142)
NordicPlayground/fw-nrfconnect-nrf: applications/nrf_desktop/README.rst

For more information about how to configure the flash memory layout using the partition manager, see :ref:`partition_manager`.

Device Firmware Upgrade
***********************

The nRF Desktop application uses background image transfer for the DFU process.
The following elements take part in the firmware update process:

>     * `Update image generation`_

Reviewer: Do not indent lists.

Author: Fixed

    * `Update image transfer`_
    * `Update image verification and swap`_

> 

Reviewer: Please add a line in line 344 to shortly describe the result of these three steps.
```
At the end of these three stages, the nRF Desktop application will be rebooted with the new firmware package installed.
```

Author: Added

Update image generation
=======================

> The update image is automatically generated if the bootloader is enabled and can be found at the application build directory.

Author: I will add more information here under subsequent PR when the switch to B0 is complete.

Author: Done

> The update image is automatically generated if the bootloader is enabled and can be found at the application build directory.

Author: Fixed.

Update image transfer
=====================

> The update image is transmitted through the :ref:`nrf_desktop_config_channel` while the device is in normal operation.

Author: You can move cursor, press buttons. In other words - application does not enter any special DFU state.

> The update image is transmitted through the :ref:`nrf_desktop_config_channel` while the device is in normal operation.

Author: Rephrased

The device must be connected to the host via the USB or the Bluetooth.

> The update image data is transmitted using HID feature reports.

Author: This is not connected with HID_state nor HIDS. All explanation is in config channel documentation. I have rephrased this text as it sound misleading.

> At the application side the process is handled by :ref:`nrf_desktop_dfu`.
> Please refer to the module documentation for how it should be enabled and configured.
> 
> At the host side the process is handled by the :ref:`nrf_desktop_config_channel_script`.
> Please refer to the tool documentation for more information how execute the DFU process on the host.

Reviewer: ```suggestion
Depending on the side on which the process is handled:

* On the application side, the process is handled by :ref:`nrf_desktop_dfu`.
  See the module documentation for how to enable and configure it.
* On the host side, the process is handled by the :ref:`nrf_desktop_config_channel_script`.
  See the tool documentation for more information about how to execute the DFU process on the host.
```

Update image verification and swap
==================================

> Once the update image transfer is completed, the DFU process will continue after the device reboot.

Reviewer: Should I trigger the reboot? Because your documentation suggests it is happening automatically.

Author: Changed this text a bit.
When you use our hid configurator the reboot will be done automatically. I am not sure any client will actually use it but we don't really know.

> The reboot can be done right after the image transfer is done.

Reviewer: ```suggestion
The reboot can be done right after the image transfer.
```

> After the reboot the bootloader will find the update image loaded on the update partition of the device.

Reviewer: ```suggestion
After the reboot, the bootloader locates the update image on the update partition of the device.
```

> The image verification process will be performed to ensure the image's integrity and to check if its signature is valid.

Reviewer: ```suggestion
The image verification process ensures the integrity of the image and checks if its signature is valid.
```
This sentence should start this section. Move it to line 365. You can then update line 368 with information about verification taking place:
```
After the reboot, the bootloader locates the update image on the update partition of the device and starts the verification procedure.
```

> If this process pass the bootloader boot the new version of the application.

Reviewer: ```suggestion
If verification is successful, the bootloader boots the new version of the application.
```
